app/components/floating_banner/floating_banner.tsx

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {StyleSheet, Platform} from 'react-native';
+import {GestureHandlerRootView} from 'react-native-gesture-handler';
+import Animated, {useAnimatedStyle, withTiming} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+import Banner from '@components/banner/Banner';
+import BannerItem from '@components/banner/banner_item';
+import {
+    FLOATING_BANNER_BOTTOM_OFFSET_PHONE_IOS,
+    FLOATING_BANNER_BOTTOM_OFFSET_PHONE_ANDROID,
+    FLOATING_BANNER_BOTTOM_OFFSET_WITH_KEYBOARD_IOS,
+    FLOATING_BANNER_BOTTOM_OFFSET_WITH_KEYBOARD_ANDROID,
+    FLOATING_BANNER_TABLET_EXTRA_BOTTOM_OFFSET,
+    CHANNEL_BANNER_HEIGHT,
+    DEFAULT_HEADER_HEIGHT,
+    TABLET_HEADER_HEIGHT,
+    BANNER_SPACING,
+} from '@constants/view';
+import {useIsTablet, useKeyboardHeight} from '@hooks/device';
+
+import type {BannerConfig} from './types';
+
+type FloatingBannerProps = {
+    banners: BannerConfig[];
+    onDismiss: (id: string) => void;
+};
+
+type BannerSectionProps = {
+    sectionBanners: BannerConfig[];
+    position: 'top' | 'bottom';
+    executeBannerAction: (banner: BannerConfig) => void;
+    dismissBanner: (banner: BannerConfig) => void;
+};
+
+type AnimatedBannerItemProps = {
+    banner: BannerConfig;
+    index: number;
+    isTop: boolean;
+    isTablet: boolean;
+    insets: {top: number; bottom: number; left: number; right: number};
+    executeBannerAction: (banner: BannerConfig) => void;
+    dismissBanner: (banner: BannerConfig) => void;
+};
+
+const AnimatedBannerItem: React.FC<AnimatedBannerItemProps> = ({
+    banner,
+    index,
+    isTop,
+    isTablet,
+    insets,
+    executeBannerAction,
+    dismissBanner,
+}) => {
+    const {id, dismissible = true, customComponent} = banner;
+
+    let baseOffset = BANNER_SPACING;
+    if (isTop) {
+        baseOffset += insets.top;
+
+        if (isTablet) {
+            baseOffset += TABLET_HEADER_HEIGHT;
+        } else {
+            baseOffset += DEFAULT_HEADER_HEIGHT;
+        }
+    }
+
+    const stackOffset = baseOffset + (index * (CHANNEL_BANNER_HEIGHT + BANNER_SPACING));
```

Comment:
based on this we are considering that every banner will have the same height as the channel banner?